tuf/ngclient/updater.py

```diff
@@ -173,59 +178,63 @@ def get_one_valid_targetinfo(
         """
         return self._preorder_depth_first_walk(target_path)
 
-    @staticmethod
-    def updated_targets(
-        targets: List[TargetFile], destination_directory: str
-    ) -> List[TargetFile]:
-        """Checks whether local cached target files are up to date
-
-        After retrieving the target information for the targets that should be
-        updated, updated_targets() can be called to determine which targets
-        have changed compared to locally stored versions.
+    def find_cached_target(
+        self,
+        targetinfo: TargetFile,
+        filepath: Optional[str] = None,
+    ) -> Optional[str]:
+        """Checks whether a local file is an up to date target
 
-        All the targets that are not up-to-date in destination_directory are
-        returned in a list. The list items can be downloaded with
-        'download_target()'.
-        """
-        # Keep track of TargetFiles and local paths. Return 'updated_targets'
-        # and use 'local_paths' to avoid duplicates.
-        updated_targets: List[TargetFile] = []
-        local_paths: List[str] = []
+        If the ``target_dir`` argument was given to constructor, ``filepath``
+        argument is not required here: in this case a filename will be
+        generated based on the target path like in ``download_target()``
 
-        for target in targets:
-            # URL encode to get local filename like download_target() does
-            filename = parse.quote(target.path, "")
-            local_path = os.path.join(destination_directory, filename)
+        Args:
+            targetinfo: TargetFile from ``get_targetinfo()``.
+            filepath: Local path to file. By default a filename is generated
+                and file is looked for in ``target_dir`` (see note above).
```

Comment:
What about this? Feels a bit less ambiguous.

```suggestion
            filepath: Local path to file. If not passed, filepath is generated using 
            `self.target_dir`, plus the url-quoted `path` from the passed `targetinfo`.
```

Reply:
yeah the intent is clearer: I don't think I'll include the actual filename algorithm here (as it's internal to updater and the final path is returned) but otherwise this makes sense.

Reply:
I've tried to incorporate this idea and removed the paragraphs in the method docstrings